### PR TITLE
Add `ReduceLROnPlateau` to `torch.lr_scheduler_resolver`

### PR DESCRIPTION
**Pull request by rusty1s**
The `ClassResolver` is unable to add `torch.optim.lr_scheduler.ReduceLROnPlateau` to the resolver because it inherits from `object` and not `torch.optim.lr_scheduler._LRScheduler`. This PR fixes this by explicitly registering it.

### CODE REVIEW

**Comment by cthoyt** [comment]
It appears that `ReduceLROnPlateau` is not available the way you imported it. Is there an alternative route? Or is this based on the pytorch version whether it's available?

**Comment by rusty1s** [comment]
Yeah, looks like I was too fast. Sorry :)

**Review by cthoyt — APPROVED** [review]
Very strange that `ReduceLROnPlateau` doesn't inherit from the base class. Thanks for pointing this out and making a PR!

**Comment by mberr** [comment]
> Very strange that `ReduceLROnPlateau` doesn't inherit from the base class. Thanks for pointing this out and making a PR!

To give some context: I think this is due to `ReduceLROnPlateau` not adhering to the general API of other learning rate schedulers, as it is the only one requiring an additional monitored metric. We already experienced something similar in this pykeen issue: https://github.com/pykeen/pykeen/pull/492#discussion_r652846931
